### DISCUSSION TITLE
#471 and #488

### DISCUSSION THREAD

**Discussion by hassankhan** [serverless-heaven/serverless-webpack#1538] [Team Posts]
Given that we seem to be getting a bunch of PRs that add configuration values, might it be a sensible idea to define a somewhat consistent format?

I only say this because seeing a bunch of options like `excludeFiles` and `builtAssetIncludeGlob` _feel_ like they could be presented in a better way, and leave the door open for adding future (similar) options in a similarly consistent way.

I could be completely wrong here, so just wanted to get your thoughts

/cc @designfrontier @HyperBrain

**Comment by designfrontier** [comment]
yeah I agree but I am not sure exactly what we would want to adopt. What are your ideas?

**Comment by HyperBrain** [comment]
Yes, especially the builtAssetIncludeGlob looks a bit clumsy. Additionally,
I don't think that name parts like "Glob" are necessary because it is more
like a specific syntax, but does not tell about the meaning of the option.
In general, we should imply that any option/filter that affects files or
the like can be expressed as Globs.

Maybe we should set the guideline as follows:
if it is for a special action on something, it should name the action and
then the target like _includeFiles_, _excludeFiles_, etc.
if it sets a property, it should be only the property like _packager_
if it configures a property (i.e. it is a collection/container of options)
like _packagerOptions_ it should name the property and then "Options". The
contained options should then NOT repeat the parent property anymore in
their names.

**Comment by hassankhan** [comment]
+1, makes sense to me, would be another nice tidbit to add to the `CONTRIBUTING.md` document.
